src/db_parser/EPICS_collections.py

```diff
@@ -0,0 +1,172 @@
+    """
+
+    def __init__(self, rec_type, pv, infos, fields, aliases):
+        self.type = rec_type
+        self.pv = pv
+        self.fields = fields
+        self.infos = infos
+        self.aliases = aliases
+        # Test for whether the PV is a simulation
+        if re.search(r'.SIM(:.|$)', pv) is not None:
+            self.simulation = True
+        else:
+            self.simulation = False
+
+        # Test for whether the PV is a disable
+        if re.search(r'DISABLE', pv) is not None:
+            self.disable = True
+        else:
+            self.disable = False
+
+    def is_sim(self):
+        return self.simulation
+
+    def is_disable(self):
+        return self.disable
+
+    def __str__(self):
+        return str(self.pv)
+
+    def get_field_names(self):
+        """
+        This method returns all field names as a list
+        """
+        return [field.name for field in self.fields]
+
+    def get_field_values(self):
+        """
+        This method returns all field values as a list
+        """
+        return [field.value for field in self.fields]
+
+    def has_field(self, search):
+        """
+        This method checks all contained fields for instances of a
+        pv given by the search input
+        """
+        for field in self.fields:
+            if field.name == search:
+                return True
+        return False
+
+    def get_field_value(self, search):
+        """
+        This method returns the values of the first field contained within
+        the record that matches the search input
+        If no field exists None is returned
+        """
+        for field in self.fields:
+            if field.name == search:
+                return field.value
+        return None
+
+    def get_field(self, search):
+        """
+        This method returns the values of the first field contained within
+        the record that matches the search input
+        If no field exists None is returned
+        """
+        for field in self.fields:
+            if field.name == search:
+                return field
+        return None
+
+    def get_type(self):
+        """
+        This method returns the PV type
+        """
+        return self.type
+
+    def get_info(self, search):
+        """
+        This method returns a list of the values of all contained info
+        fields that match the search input
+        """
+        found_values = []
+        for info in self.infos:
+            if info.name == search:
+                found_values.append(info.value)
+        return found_values
+
+    def is_interest(self):
+        """
+        This method returns whether or not the record is of interest
+        """
+        for info in self.infos:
+            if info.name == "INTEREST":
+                return True
+        return False
+
+    def get_siml(self):
+        for field in self.fields:
+            if field.name == "SIML":
+                return field.value
+        return None
```

Comment:
Nit picking:
```suggestion
        return [field.value if field.name == "SIML" else None for fieldin self.fields]
```
and for similar functions below

Reply:
this is currently mimicing functionality from the old parser that is needed for the add_sim_records script, and only needs to find if the value of the first field with the name SIML. the old functionality didn't care if there were multiple fields named this, so is expecting a single value return rather than a list of values

Reply:
Fair enough, I just like list comprehension :(

Reply:
so do I, but it seems silly to do a list comprehension if we only care about the first case?

Reply:
What about `next((field.value for field in self.fields if field.name =="SIML"), None)`?